wrangler-service/src/main/java/co/cask/wrangler/service/kafka/KafkaService.java

```diff
@@ -258,19 +284,89 @@ public void read(HttpServiceRequest request, HttpServiceResponder responder,
         object.addProperty(PropertyIds.ID, uuid);
         object.addProperty(PropertyIds.NAME, topic);
         values.add(object);
-        response.addProperty("status", HttpURLConnection.HTTP_OK);
-        response.addProperty("message", "Success");
-        response.addProperty("count", values.size());
-        response.add("values", values);
+        response.addProperty(PropertyIds.STATUS, HttpURLConnection.HTTP_OK);
+        response.addProperty(PropertyIds.MESSAGE, PropertyIds.SUCCESS);
+        response.addProperty(PropertyIds.COUNT, values.size());
+        response.add(PropertyIds.VALUES, values);
         sendJson(responder, HttpURLConnection.HTTP_OK, response.toString());
-      } finally {
-        consumer.close();
-      }
-    } catch (Exception e) {
+    } catch (Throwable e) {
+      LOG.error("Exception while reading data from kafka topic", e);
       error(responder, e.getMessage());
     }
   }
+  
+  
+  
+  public void generateSchema(KafkaConfiguration config, String format) throws Exception {
```

Comment:
Its always better to return value instead of functions returning void. We can return outputSchema here. This method should be 'private'.